doc/dune-files.rst

```diff
@@ -327,6 +327,12 @@ OCaml/Reason modules only consume modules from the directory where the
 stanza appear. In order to declare a multi-directory library, you need
 to use the :ref:`include_subdirs` stanza.
 
+Note also that a nested library in a project is not automatically namespaced
+under its parent library. E.g. if you have a public library ``foo`` which nests
+another public library ``bar``, it will not be exposed as ``Foo.Bar`` outside
+the project, but as ``Bar``. Take care to give unique names to nested libraries
+to avoid conflicts. E.g., ``foo__Bar``.
```

Comment:
double underscores in names are a (private) artifact of how dune builds modules, I don't think it's a good idea to reuse this kind of names.

Reply:
Disagree, double underscores are a widespread convention in the OCaml community which dune adopted. Dune doesn't 'own' the double underscores naming convention.